src/apify/_charging.py

```diff
@@ -328,6 +334,54 @@ def get_charged_event_count(self, event_name: str) -> int:
     def get_max_total_charge_usd(self) -> Decimal:
         return self._max_total_charge_usd
 
+    async def _fetch_pricing_info(self) -> dict[str, Any]:
+        """Fetch pricing information from environment variables or API."""
+        # Check if pricing info is available via environment variables
+        if self._configuration.actor_pricing_info and self._configuration.charged_event_counts:
+            charged_counts = json.loads(self._configuration.charged_event_counts)
+
+            # Validate pricing info with proper discriminator support
+            pricing_info_adapter: TypeAdapter[
+                FreeActorPricingInfo
+                | FlatPricePerMonthActorPricingInfo
+                | PricePerDatasetItemActorPricingInfo
+                | PayPerEventActorPricingInfo
+            ] = TypeAdapter(
+                FreeActorPricingInfo
+                | FlatPricePerMonthActorPricingInfo
+                | PricePerDatasetItemActorPricingInfo
+                | PayPerEventActorPricingInfo
+            )
+            pricing_info = pricing_info_adapter.validate_json(self._configuration.actor_pricing_info)
```

Comment:
I'd move this to the `Configuration` class itself so that `self._configuration.actor_pricing_info` is already parsed and validated

```py
class Configuration
  ...
  actor_pricing_info: Annotated[FreeActorPricingInfo | ..., Field(...), BeforeValidator(json.loads)]
```